initfiles/sbin/complete-uninstall.sh.in

```diff
@@ -100,54 +100,60 @@ if [ -n "${lpath}" -a -h "${lpath}" ] ; then
 fi
 
 if [ -e /etc/debian_version ]; then
-    echo "Removing DEB"
-    apt-get remove -y hpccsystems-platform
-    if [ $? -ne 0 ]; then
-       echo "An error has occured during removal.  Try \`apt-get -f remove hpccsystems-platform\` to fix this issue."
+    pckname=$(apt-cache pkgnames hpccsystems-platform)
+    if [ -n "${pckname}" ]; then
+        echo "Removing ${pckname}"
+        apt-get remove -y ${pckname}
+    elif [ $? -ne 0 ]; then
+       echo "An error has occured during removal.  Try \`apt-get -f remove ${pckname}\` to fix this issue."
        exit 1
     fi
 elif [ -e /etc/redhat-release -o -e /etc/SuSE-release ]; then
-    echo "Removing RPM"
-    yum remove -y hpccsystems-platform
-    if [ $? -ne 0 ]; then
+    pckname=$(rpm -qa | grep hpccsystems-platform)
+    if [ -n "${pckname}" ]; then
+        echo "Removing ${pckname}"
+        yum remove -y ${pckname}
+    elif [ $? -ne 0 ]; then
        echo "An error has occured during removal.  Try to fix any issues and then run \`yum-complete-transaction\`."
        exit 1
     fi
 fi
 
-echo "Removing Directory - ${path}"
-removedir ${path}
+# Removing directories
+if [ -n "${pckname}" ]; then
```

Comment:
I would reverse this and just exit here if pckname was empty (-z) so we didn't have to edit so many lines below.